api/_db.data-utils.js

```diff
@@ -55,6 +55,27 @@ const parseDate = (rawValue) => {
 	return parsed;
 };
 
+const findHighestValidCertificate = (certificates) => {
+	// find the certificate that is not expired and has the highest value (red is higher than green).
+	// nonexpired green has to beat expired red, but red with shorter expiryTime has to beat any green.
+	return certificates.reduce((previous, current) =>{
+		if (!previous) {
+			return current;
+		}
+		if (parseDate(current.expiryTime) > Date.now()){
+			// TODO: maybe we should find the certificate with the longest expiry time
```

Comment:
I think we should find the highest-degree **valid** certificate.
If multiple found, the last issued valid certificate.
And if that not found, the last issued invalid certificate.